src/quick-lint-js/diag/diagnostic-types.h

```diff
@@ -1751,6 +1751,18 @@
               question)                                                         \
           MESSAGE(QLJS_TRANSLATABLE("initializer starts here"), equal))         \
                                                                                 \
+  QLJS_DIAG_TYPE(                                                               \
+      diag_optional_parameter_cannot_be_followed_by_required_parameter,         \
+      "E0375", diagnostic_severity::error,                                      \
+      {                                                                         \
+        source_code_span optional_parameter;                                    \
+        source_code_span required_parameter;                                    \
+      },                                                                        \
+      MESSAGE(QLJS_TRANSLATABLE("optional parameter"), optional_parameter)      \
+          MESSAGE(QLJS_TRANSLATABLE("cannot be followed by a required "         \
+                                    "parameter"),                               \
+                  required_parameter))                                          \
```

Comment:
Must fix: In our diagnostics, we assume that the second message might not be visible. (Some editors don't show the second message, or only show them if you hover.) The first message should be understandable on its own.

For example:
```suggestion
      MESSAGE(QLJS_TRANSLATABLE("optional parameter cannot be followed by a required parameter"), optional_parameter)      \
          MESSAGE(QLJS_TRANSLATABLE("this required parameter appears after the optional parameter"),         \
                  required_parameter))                                          \
```